### PR TITLE
CI Improvements

### PR DESCRIPTION
**Pull request by whereismyjetpack**
- switches to multi-stage builds for baking test targets
- run test targets on those stages
- parallelize testing stages

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -22,41 +27,82 @@ steps:
...
+    - export DOCKER_HOST=unix:///var/run/host.sock
+    - apk add bind-tools
+    - export POSTGRES_HOST=$(dig +short postgres)
+    - export MINIO_HOST=$(dig +short minio)
+    - docker run
+        --net host
+        -e POSTGRES_PASSWORD=scholarsphere
+        -e AWS_BUCKET=scholarsphere
+        -e S3_ENDPOINT=http://$MINIO_HOST:9000
+        -e AWS_BUCKET=scholarsphere
```
Comment (awead) [inline .drone.yml]: duplicate line.

File: .drone.yml [hunk 2]
```diff
@@ -22,41 +27,82 @@ steps:
...
-    - bundle exec rails db:create
-    - bundle exec rails db:migrate
-    - bin/ci-rspec
+    - export DOCKER_HOST=unix:///var/run/host.sock
+    - apk add bind-tools
+    - export POSTGRES_HOST=$(dig +short postgres)
+    - export MINIO_HOST=$(dig +short minio)
+    - docker run
+        --net host
+        -e POSTGRES_PASSWORD=scholarsphere
```
Comment (awead) [inline .drone.yml]: should this be hidden or do we care? I'm not sure I do :)

File: .drone.yml (hunk 1 above)
Comment (whereismyjetpack) [reply .drone.yml]: good eye, sniper.

File: .drone.yml (hunk 2 above)
Comment (whereismyjetpack) [reply .drone.yml]: it's ephemerial, i don't think we care.

Comment (awead) [reply .drone.yml]: since the host name is hidden this probably doesn't matter?

**Pull request merged**